apps/nse-vlan-vpp/nse.yaml

```diff
@@ -0,0 +1,57 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nse-kernel
+  labels:
+    app: nse-kernel
+spec:
+  selector:
+    matchLabels:
+      app: nse-kernel
+  template:
+    metadata:
+      labels:
+        app: nse-kernel
+        "spiffe.io/spiffe-id": "true"
+    spec:
+      containers:
+        - name: nse
+          image: ghcr.io/networkservicemesh/ci/cmd-nse-vlan-vpp:88df38a
+          imagePullPolicy: IfNotPresent
+          securityContext:
+            privileged: true
+          env:
+            - name: SPIFFE_ENDPOINT_SOCKET
+              value: unix:///run/spire/sockets/agent.sock
+            - name: NSM_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: NSM_LOG_LEVEL
+              value: TRACE
+            - name: NSM_CONNECT_TO
+              value: unix:///var/lib/networkservicemesh/nsm.io.sock
+          volumeMounts:
+            - name: spire-agent-socket
+              mountPath: /run/spire/sockets
+              readOnly: true
+            - name: nsm-socket
+              mountPath: /var/lib/networkservicemesh
+              readOnly: true
+          resources:
+            requests:
+              cpu: 1
+              memory: 4Gi
```

Comment:
That's a huge amount of memory... why so much?

Reply:
@edwarnicke I've seen vpp crash with below message for lesser memory , so had set to 4Gi to avoid this crash. 
Let me have a look at it again.

```
Nov 16 09:19:30.978 [INFO] [cmd:vpp] vpp[12]: buffer: numa[1] falling back to non-hugepage backed buffer pool (vlib_physmem_shared_map_create: pmalloc_map_pages: Unable to lock pages: Cannot allocate memory)
2021/11/16 09:19:33 [ERROR] signal: killed
```

Reply:
```
Nov 16 09:19:30.978 [INFO] [cmd:vpp] vpp[12]: buffer: numa[1] falling back to non-hugepage backed buffer pool (vlib_physmem_shared_map_create: pmalloc_map_pages: Unable to lock pages: Cannot allocate memory)
```
Seems to be about 3 seconds before the 

```
2021/11/16 09:19:33 [ERROR] signal: killed
```

```
Nov 16 09:19:30.978 [INFO] [cmd:vpp] vpp[12]: buffer: numa[1] falling back to non-hugepage backed buffer pool (vlib_physmem_shared_map_create: pmalloc_map_pages: Unable to lock pages: Cannot allocate memory)
```
is just VPP complaining that it really wanted hugepages before moving on with life... I believe we see it all the time.

```
2021/11/16 09:19:33 [ERROR] signal: killed
```

Looks like a kill signal was received. I don't think that log message is from VPP though... I suspect its from your Pod being killed by kubelet for some reason.

So the question is... whats the root cause.  What have you observed about the memory consumption of the processes in the Pod?

I am seeing indications [here](https://my-vpp-docs.readthedocs.io/en/latest/gettingstarted/users/configuring/startup.html) that the ```heap-size``` defaults to 1Gb... its possible that we are growing to be the biggest Pod on a Node with memory pressure and getting killed off.

Reply:
@edwarnicke Thanks for your inputs. yes, the error log not from vpp itself but I think the error is passed via `vppErrCh` channel and logged [here](https://github.com/networkservicemesh/cmd-nse-vlan-vpp/blob/main/main.go#L369) before closing the context and later on pod is killed by kubelet due to OOM error.

The vpp is the top memory/cpu consuming process on my single node cluster. Here is the log:
```
# ps -eo pid,ppid,cmd,%mem,%cpu --sort=-%mem | head
  PID  PPID CMD                         %MEM %CPU
57051 57030 vpp -c /etc/vpp/helper/vpp.  1.6  2.3
4256  4171 kube-apiserver --advertise-  0.4  6.1
  944     1 /usr/lib/systemd/systemd-jo  0.2  0.4
24703     1 /usr/bin/dockerd --add-runt  0.1  4.9
4847     1 /usr/bin/kubelet --bootstra  0.1  7.9
6444     1 ovs-vswitchd unix:/run/open  0.1  1.5
4265  4178 kube-controller-manager --a  0.1  1.1
7808  7786 /opt/spire/bin/spire-server  0.0  194
4272  4189 etcd --advertise-client-url  0.0  2.3
```

I've tried to use `VPP_VERSION=v22.02-rc0-100-gac6dd7c7f` docker image and `github.com/edwarnicke/vpphelper v0.0.0-20210225052320-b4f1f1aff45d` as similar to `cmd-nse-icmp-responder-vpp` repo for this example, but still seeing the same issue. 

I've also tried to bringup other vpp [example](https://github.com/networkservicemesh/deployments-k8s/tree/main/examples/use-cases/Memif2Memif), same error occurs.

Should I disable hugepage configuration on the host and try it ?

Reply:
Ah... I missed that you had hugepages enabled!

In principle it shouldn't matter... but if it's easy for you to try with hugepages disabled it would be good to find out if you can reproduce in that situation.

When you see the issue at lower memory thresholds, is it still true that limit is set to the same value as request?

Also: what's the absolute amount of memory your vpp is consuming around the time it is killed?  I ask, because it should be the case that the VPP heap by default is set to 1G

Reply:
@edwarnicke when I ran this example on a cluster (VM) without hugepage config, then nse just needed 214M for vpp process whereas it needed 2.2G when worker had hugepage config. how do you want to go about this ? please suggest.

Reply:
@pperiyasamy That's interesting.  Did the VPP instance *get* any of those huge pages in the case where they were configured?

Reply:
No Ed, The nse container was not started with a hugepage resource and vpp doesn't also create any huge page mmap files. this 2.2G is just a resident memory used by vpp process.

Reply:
Got it... OK... so that's an issue in VPP to also explore.

@pperiyasamy How would you suggest resolving the question here in your PR?

Reply:
@edwarnicke I will change the pod spec to use lesser memory/cpu with a note (in readme) saying nse container would need more memory when host has set with hugepage.

Reply:
@pperiyasamy That sounds perfect :)

Reply:
done.